src/Service/UserTransformer.php

```diff
@@ -90,4 +105,34 @@ public function transform(
         }
         return $entity;
     }
+
+    /**
+     * @param UserDTO $dto
+     * @return User
+     * @throws NotValidOldPassword
+     * @throws EntityNotFound
+     */
+    public function changePasswordTransform(
+        UserDTO $dto,
+        $id
```

Comment:
Send the User object, not the id. You already searched the user in the database, there's no reason to do it twice.